src/command_executor.cc

```diff
@@ -75,30 +87,40 @@ void CommandExecutor::WaitChildExit(pid_t pid) {
 }
 
 int CommandExecutor::Execute(const std::string &line) {
-  CommandParseResult command_parse_result =
+  std::vector<CommandParseResult> command_parse_result_list =
       parser_.ParseUserInputLine(line);
-  if (build_in_.Exist(command_parse_result.command)) {
-    return build_in_.Execute(command_parse_result.command,
-                             command_parse_result.args);
-  }
+  bool is_pipe_redirect = command_parse_result_list.size() != 1;
+  int pipe_fds[2];
+  pipe(pipe_fds);
```

Comment:
I notice that function pipe return int. What does it means for this int value? Do we need to handle it?`int	 pipe(int [2]);`